pkg/autoscaler/autoscaler_test.go

```diff
@@ -113,3 +114,141 @@ func TestAutoscaler_Awake(t *testing.T) {
 		return latestRequest != 1
 	}, downscaleStabilizationPeriod, time.Second)
 }
+
+func TestAutoscaler_MinReplicas(t *testing.T) {
+	t.Parallel()
+	log := newLogger(t)
+
+	mux := sync.RWMutex{}
+	var latestRequest int32
+
+	minReplicas := int32(5)
+	maxReplicas := int32(10)
+
+	scalerMock := &ScalerFunc{
+		ScaleFunc: func(apiName string, request int32) error {
+			mux.Lock()
+			defer mux.Unlock()
+
+			latestRequest = request
+			return nil
+		},
+		GetInFlightRequestsFunc: func(apiName string, window time.Duration) (*float64, error) {
+			return pointer.Float64(0), nil
+		},
+		GetAutoscalingSpecFunc: func(apiName string) (*userconfig.Autoscaling, error) {
+			return &userconfig.Autoscaling{
+				MinReplicas:    minReplicas,
+				MaxReplicas:    maxReplicas,
+				InitReplicas:   minReplicas,
+				TargetInFlight: pointer.Float64(1),
+				Window:         time.Second,
```

Comment:
Is it okay if the stabilization periods are set to zero? Same for downscale and upscale factors? This would also apply for the other test.

Reply:
yes, for the stabilization periods. Not sure about the downscale and upscale factors

Reply:
updated the scale factors to the default ones